data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapper.java

```diff
@@ -21,18 +21,22 @@
 class GrokLogstashPluginAttributesMapper extends AbstractLogstashPluginAttributesMapper {
     protected static final String LOGSTASH_GROK_MATCH_ATTRIBUTE_NAME = "match";
     protected static final String LOGSTASH_GROK_PATTERN_DEFINITIONS_ATTRIBUTE_NAME = "pattern_definitions";
+    protected static final String LOGSTASH_GROK_OVERWRITE_ATTRIBUTE_NAME = "overwrite";
     private static final Logger LOG = LoggerFactory.getLogger(GrokLogstashPluginAttributesMapper.class);
 
     @SuppressWarnings("unchecked")
     @Override
     protected void mapCustomAttributes(List<LogstashAttribute> logstashAttributes, LogstashAttributesMappings logstashAttributesMappings, Map<String, Object> pluginSettings) {
         final List<LogstashAttribute> matchAttributes = new ArrayList<>();
         final Map<String, String> patternDefinitions = new HashMap<>();
+        final List<String> keysToOverwrite = new ArrayList<>();
         logstashAttributes.forEach(logstashAttribute -> {
             if (logstashAttribute.getAttributeName().equals(LOGSTASH_GROK_MATCH_ATTRIBUTE_NAME)) {
                 matchAttributes.add(logstashAttribute);
             } else if (logstashAttribute.getAttributeName().equals(LOGSTASH_GROK_PATTERN_DEFINITIONS_ATTRIBUTE_NAME)) {
                 patternDefinitions.putAll((Map<String, String>) logstashAttribute.getAttributeValue().getValue());
+            } else if (logstashAttribute.getAttributeName().equals(LOGSTASH_GROK_OVERWRITE_ATTRIBUTE_NAME)) {
+                keysToOverwrite.addAll((List<String>) logstashAttribute.getAttributeValue().getValue());
             }
```

Comment:
This class is doing multiple things. Consider moving the responsibility of each if condition to separate class to make this class more maintainable.

Reply:
Do you mean extract this into a method? Creating a new class for this would be confusing for plugin authors for adding converter logic. I can extract it to a separate method.